osbs/build/build_requestv2.py

```diff
@@ -0,0 +1,225 @@
+        # all plugins that might cause importing of data to koji
+        self.scratch = kwargs.get('scratch')
+        # When true, it indicates build was automatically started by
+        # OpenShift via a trigger, for instance ImageChangeTrigger
+        self.is_auto = kwargs.pop('is_auto', False)
+        # An isolated build is meant to patch a certain release and not
+        # update transient tags in container registry
+        self.isolated = kwargs.get('isolated')
+
+        self.validate_build_variation()
+
+        self.base_image = kwargs.get('base_image')
+        self.platform_node_selector = kwargs.get('platform_node_selector', {})
+        self.scratch_build_node_selector = kwargs.get('scratch_build_node_selector', {})
+        self.explicit_build_node_selector = kwargs.get('explicit_build_node_selector', {})
+        self.auto_build_node_selector = kwargs.get('auto_build_node_selector', {})
+        self.isolated_build_node_selector = kwargs.get('isolated_build_node_selector', {})
+
+        logger.debug("setting params '%s' for %s", kwargs, self.user_params)
+        self.user_params.set_params(**kwargs)
+
+    # Override
+    @property
+    def inner_template(self):
+        raise RuntimeError('inner_template not supported in BuildRequestV2')
+
+    # Override
+    @property
+    def customize_conf(self):
+        raise RuntimeError('customize_conf not supported in BuildRequestV2')
+
+    # Override
+    @property
+    def dj(self):
+        raise RuntimeError('DockJson not supported in BuildRequestV2')
+
+    def set_reactor_config_map(self, api):
+        if not self.user_params.reactor_config_map.value:
+            return
+        super(BuildRequestV2, self).set_reactor_config_map(self.user_params.reactor_config_map)
+        cm_response = api.get_config_map(self.user_params.reactor_config_map.value)
+        custom = self.template['spec']['strategy']['customStrategy']
+        custom.setdefault('secrets', [])
+        existing = [secret_mount['secretSource']['name'] for secret_mount in custom['secrets']]
+        for key in _CM_SECRETS:
+            secrets = cm_response.get_data_by_key(key)
+            for secret in secrets:
+                if secret in existing:
+                    continue
+                else:
+                    custom['secrets'].append({
+                        'secretSource': {
+                            'name': secret,
+                        },
+                        'mountPath': os.path.join(SECRETS_PATH, secret),
+                    })
+
+    def adjust_for_scratch(self):
+        """
+        Scratch builds must not affect subsequent builds,
+        and should not be imported into Koji.
+        """
+        if self.scratch:
+            self.template['spec'].pop('triggers', None)
+            self.set_label('scratch', 'true')
+
+    def render_node_selectors(self):
+        # for worker builds set nodeselectors
+        if self.user_params.platforms.value is None:
```

Comment:
Change parent class to take `platforms` as parameter to avoid duplication?

Reply:
Sure, fixed.